Fix warnings for unused local typedef

The following warnings appear in release mode:
```
./source/hp/dof_handler.cc:602:44: warning: unused typedef 'BaseClass' [-Wunused-local-typedef]
          typedef DoFHandler<dim,spacedim> BaseClass;
                                           ^
./source/hp/dof_handler.cc:628:44: warning: unused typedef 'BaseClass' [-Wunused-local-typedef]
          typedef DoFHandler<dim,spacedim> BaseClass;
                                           ^
./source/hp/dof_handler.cc:658:44: warning: unused typedef 'BaseClass' [-Wunused-local-typedef]
          typedef DoFHandler<dim,spacedim> BaseClass;
                                           ^
```

--- a/source/hp/dof_handler.cc
+++ b/source/hp/dof_handler.cc
@@ -599,12 +599,10 @@ namespace internal
         {
           const unsigned int dim = 1;
 
-          typedef DoFHandler<dim,spacedim> BaseClass;
-
           Assert (dof_handler.finite_elements != nullptr,
-                  typename BaseClass::ExcNoFESelected());
+                  (typename DoFHandler<dim,spacedim>::ExcNoFESelected()));
           Assert (dof_handler.finite_elements->size() > 0,
-                  typename BaseClass::ExcNoFESelected());
+                  (typename DoFHandler<dim,spacedim>::ExcNoFESelected()));
           Assert (dof_handler.tria->n_levels() > 0,
                   ExcMessage("The current Triangulation must not be empty."));
           Assert (dof_handler.tria->n_levels() == dof_handler.levels.size (),
@@ -625,12 +623,10 @@ namespace internal
         {
           const unsigned int dim = 2;
 
-          typedef DoFHandler<dim,spacedim> BaseClass;
-
           Assert (dof_handler.finite_elements != nullptr,
-                  typename BaseClass::ExcNoFESelected());
+                  (typename DoFHandler<dim,spacedim>::ExcNoFESelected()));
           Assert (dof_handler.finite_elements->size() > 0,
-                  typename BaseClass::ExcNoFESelected());
+                  (typename DoFHandler<dim,spacedim>::ExcNoFESelected()));
           Assert (dof_handler.tria->n_levels() > 0,
                   ExcMessage("The current Triangulation must not be empty."));
           Assert (dof_handler.tria->n_levels() == dof_handler.levels.size (),
@@ -655,12 +651,10 @@ namespace internal
         {
           const unsigned int dim = 3;
 
-          typedef DoFHandler<dim,spacedim> BaseClass;
-
           Assert (dof_handler.finite_elements != nullptr,
-                  typename BaseClass::ExcNoFESelected());
+                  (typename DoFHandler<dim,spacedim>::ExcNoFESelected()));
           Assert (dof_handler.finite_elements->size() > 0,
-                  typename BaseClass::ExcNoFESelected());
+                  (typename DoFHandler<dim,spacedim>::ExcNoFESelected()));
           Assert (dof_handler.tria->n_levels() > 0,
                   ExcMessage("The current Triangulation must not be empty."));
           Assert (dof_handler.tria->n_levels() == dof_handler.levels.size (),

